plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2206,101 +2180,105 @@ void read_write::push_transactions(const read_write::push_transactions_params& p
+                     if( retry && api.trx_retry.has_value() && !trx_trace_ptr->except) {
+                        // will be ack'ed via next later
+                        api.trx_retry->track_transaction( ptrx, retry_num_blocks,
+                             [ptrx, next](const next_function_variant<std::unique_ptr<fc::variant>>& result ) {
+                                if( std::holds_alternative<fc::exception_ptr>( result ) ) {
+                                   next( std::get<fc::exception_ptr>( result ) );
+                                } else {
+                                   fc::variant& output = *std::get<std::unique_ptr<fc::variant>>( result );
+                                   next( Result{ptrx->id(), std::move( output )} );
+                                }
+                             } );
+                        retried = true;
                      }
-                     const chain::transaction_id_type& id = trx_trace_ptr->id;
-                     next( read_write::send_transaction_results{id, std::move( output )} );
+                  }
+
+                  if (!retried) {
+                     // we are still on main thread here. The lambda passed to `next()` below will be executed on the http thread pool
+                     auto yield        = abi_serializer::create_yield_function(fc::microseconds::maximum());
+                     auto abi_cache    = abi_serializer_cache_builder(api_base::make_resolver(api.db, std::move(yield))).add_serializers(trx_trace_ptr).get();
+                     using return_type = t_or_exception<Result>;
+                     next([&api, trx_trace_ptr, resolver = abi_resolver(std::move(abi_cache))]() mutable {
+                        try {
+                           fc::variant output;
+                           try {
+                              abi_serializer::to_variant(*trx_trace_ptr, output, std::move(resolver),
+                                                         abi_serializer::create_yield_function(api.abi_serializer_max_time));
+                           } catch( abi_exception& ) {
+                              output = *trx_trace_ptr;
+                           }
+                           const transaction_id_type& id = trx_trace_ptr->id;
+                           return return_type(Result{id, std::move( output )});
+                        } CATCH_AND_RETURN(return_type);
+                     });
                   }
                } CATCH_AND_CALL( next );
             }
          });
    } catch ( boost::interprocess::bad_alloc& ) {
-      chain_plugin::handle_db_exhaustion();
+      handle_db_exhaustion();
    } catch ( const std::bad_alloc& ) {
-      chain_plugin::handle_bad_alloc();
+      handle_bad_alloc();
    } CATCH_AND_CALL(next);
 }
+   
+void read_write::send_transaction(const read_write::send_transaction_params& params, next_function<read_write::send_transaction_results> next) {
```

Comment:
This should be by value. It is called via a move I think. Otherwise, your move below is ignored and it is copied.

Reply:
What should be by value? I'm not following?

Reply:
`read_write::send_transaction_params params`

Reply:
The params?

Reply:
yes

Reply:
done